deploy/awstools/awstools.py

```diff
@@ -344,19 +344,34 @@ def auto_create_bucket(userbucketname):
             rootLogger.critical(repr(exc))
             assert False
 
-def subscribe_to_firesim_topic(email):
-    """ Subscribe a user to their FireSim SNS topic for notifications. """
+def get_snsname_arn():
+    """ If the Topic doesn't exist create it, send catch exceptions while creating. Or if it exists get arn """
     client = boto3.client('sns')
 
     aws_resource_names_dict = aws_resource_names()
     snsname = aws_resource_names_dict['snsname']
 
-    # this will either create the topic, if it doesn't exist, or just get the arn
-    response = client.create_topic(
-        Name=snsname
-    )
-    arn = response['TopicArn']
+    response = None
+    try: # this will either create the topic, if it doesn't exist, or just get the arn
+        response = client.create_topic(
+            Name=snsname
+        )
+    except client.exceptions.ClientError as err:
+        if 'AuthorizationError' in repr(err): 
+            rootLogger.critical("You don't have permissions to perform \"Topic Creation \". Required to send you email notifications. Please contact your IT administrator")
+
+        else:
+            rootLogger.critical("Unknown exception is encountered while trying to perform \"Topic Creation\"")
+
+        rootLogger.critical(err)
+        assert False
+        
+    return response['TopicArn']
+
+def subscribe_to_firesim_topic(email):
+    """ Subscribe a user to their FireSim SNS topic for notifications. """
 
+    arn = get_snsname_arn()
     response = client.subscribe(
         TopicArn=arn,
         Protocol='email',
```

Comment:
@rahulb10 this is the the call I think we should put into a another try/except block with similar structure to what we've done in `get_snsname_arn` except we will want to say they are missing "sns:Subscribe" permission.

Looking down here, if we make the `get_snsname_arn()` AuthError failure non fatal, we need to either reraise an exception that can be caught by the callers or return `None` so the callers don't make further AWS requests with a bogus arn.

Reply:
I have to still retain `assert False` in get_snsname_arn() if at all I use try/except block in `subscribe_to_firesim_topic()` . In that case for it to remain non-fatal I have to use try/except block in all the functions that call `get_snsname_arn()` including the check we are having at the beginning of buildafi.

Reply:
This is also resolved in b25cd16